migrations/extract-read-status.js

```diff
@@ -0,0 +1,87 @@
+const db = require('../db'),
+      async = require('async'),
+      BATCH_SIZE = 100;
+
+const getDbName = username => `medic-user-${username}-meta`;
+
+const createReadStatusDoc = record => {
+  const type = record.form ? 'report' : 'message';
+  const id = `read:${type}:${record._id}`;
+  return { _id: id };
+};
+
+const createDb = (username, dbName, callback) => {
+  db.db.create(dbName, err => {
+    if (err) {
+      return callback(err);
+    }
+    const params = {
+      db: dbName,
+      path: '/_security',
+      method: 'put',
+      body: {
+        admins: { names: [ username ], roles: [] },
+        members: { names: [], roles:[] }
+      }
+    };
+    db.request(params, callback);
+  });
+};
+
+const ensureDbExists = (username, dbName, callback) => {
+  db.db.get(dbName, err => {
+    if (err && err.statusCode === 404) {
+      return createDb(username, dbName, callback);
+    }
+    callback();
```

Comment:
Yeah. I'm not sure what that `err` would be at this stage if not a 404, but I've modified the code to pass it to the callback so at least we'll see it reported.